streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/KTableKTableJoinNode.java

```diff
@@ -29,18 +30,18 @@
 /**
  * Too much specific information to generalize so the KTable-KTable join requires a specific node.
  */
-public class KTableKTableJoinNode<K, V1, V2, VR> extends BaseJoinProcessorNode<K, V1, V2, VR> {
+public class KTableKTableJoinNode<K, V1, V2, VR, R> extends BaseJoinProcessorNode<K, V1, V2, VR> {
```

Comment:
\cc @bbejeck Can you have a look here? You know the optimizer code best.

Reply:
@ijuma I looked into this in more detail. We don't need a new type `R` -- instead, we should update to
```
public class KTableKTableJoinNode<K, V1, V2, VR> extends BaseJoinProcessorNode<K, Change<V1>, Change<V2>, Change<VR>>
```
and use `private final MaterializedInternal<K, VR, KeyValueStore<Bytes, byte[]>> materializedInternal;` (ie, `VR` instead of `R`. (Note, the you always pass `<..., Change<X>, X>` in the current PR, what is redundant and can be avoided.)

Reply:
@mjsax That doesn't work, it's what I tried first. As far as I can tell, the existing code was wrong but it doesn't matter due to the way generics are used here.

Reply:
@mjsax Good suggestion, I've added a commit. Let me know what you think.

Reply:
the changes to the optimizer code LGTM